docs/components/progress-bars.html

```diff
@@ -0,0 +1,121 @@
+---
+layout: page
+title: Progress Bars
+description: A component that visually communicates the completion of a task or operation. It shows the task's progress and remainder.
+---
+<section class="stacks-section">
+    {% header {element: 'h2', text: 'Base Style'} %}
+    <p class="stacks-copy">The progress bars use background gradients for the fill color. In order to properly reveal the changing color of the bar, the progress "fill" is the actual background gradient. The actual bar is a gray mask that slowly shrinks to the right to reveal the background gradient underneath.</p>
+    <div class="stacks-preview">
+{% highlight html linenos %}
+<div class="s-progress">
+    <span class="s-progress--bar" role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="33" style="width: calc(100% - 33%)"></span>
```

Comment:
Is there a way to ship this without modifying the style directly?

Reply:
Not that I know of. This is based on the current implementation of progress bars where the bar is updated on page load and the width value is calculated in Razor.

Reply:
I would say this is a valid use for inline styles. Moving this to classes would be pretty awkward, especially if the value changes dynamically via JS.

However, can we make it so that we really only set the percentage as a width, not 100%-x via calc? E.g. by putting the gradient as the background on the `s-progress`, having `s-progress-bar` with a transparent background, and `s-progress--fill` taking up the remaining space on the right (via flexbox)?

Actually, the fill could just be an `::after`, so you would only need one child element.

Reply:
Oh that's an interesting approach. You might not need any child elements at all then. 🤔 

Let me try that out.

Reply:
I don't think you can get rid of the one child element, since you can't set an inline style on a pseudo-element and thus couldn't set the width.